src/components/WeatherWidgetDropdown.vue

```diff
@@ -0,0 +1,200 @@
+        v-model="searchValue"
+        type="text"
+        class="weather-widget-dropdown__input"
+        placeholder="Search a location"
+        aria-label="Search a location for weather forecast"
+        @input="handleInputChange"
+      />
+      <ul class="weather-widget-dropdown__list">
+        <WeatherWidgetMessage
+          v-if="statusMessages.length"
+          :messages="statusMessages"
+        />
+        <template v-else-if="autocompleteLocations">
+          <li
+            v-for="location in autocompleteLocations"
+            :key="location.key"
+            class="weather-widget-dropdown__list-item"
+            @click="handleLocationChange(location)"
+          >
+            {{ `${location.localizedName}, ${location.country}` }}
+          </li>
+        </template>
+      </ul>
+    </div>
+    <button
+      v-if="selectedLocation"
+      class="weather-widget-dropdown__reset-btn"
+      @click="handleLocationChange(null)"
+    >
+      Reset
+    </button>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { computed, ref } from 'vue';
+
+import { useFetch } from '@/composables/useFetch';
+import { API_URL, ENDPOINTS } from '@/constants';
+import { mapAutocompleteLocations } from '@/utils/dataMappers';
+import { debounce } from '@/utils/debounce';
+import type { AutocompleteLocationsMapperOutput, LocationInfo } from '@/types';
+
+import WeatherWidgetMessage from './WeatherWidgetMessage.vue';
+
+const searchValue = ref('');
+
+defineProps<{ selectedLocation: LocationInfo | null }>();
+
+const emit = defineEmits<{
+  (e: 'locationChange', autocompleteLocations: LocationInfo | null): void;
+}>();
+
+const {
+  data: autocompleteLocations,
+  error: autocompleteError,
+  isLoading: isLoadingAutocomplete,
+  fetchData: fetchAutocompleteResults,
+} = useFetch<AutocompleteLocationsMapperOutput>(
+  `${API_URL}${ENDPOINTS.autocomplete}`,
+  mapAutocompleteLocations
+);
+
+const loadingMessage = computed(() =>
+  isLoadingAutocomplete.value ? 'Loading...' : ''
+);
+
+const noResultsMessage = computed(() => {
+  if (!loadingMessage.value && !autocompleteLocations.value?.length) {
+    return 'No results.';
+  }
+});
+
+const statusMessages = computed(() => {
+  return [
+    loadingMessage.value,
+    autocompleteError.value,
+    noResultsMessage.value,
+  ].filter(Boolean) as (string | Error)[];
+});
+
+const debouncedFetchAutocompleteResults = debounce(fetchAutocompleteResults);
+
+const handleInputChange = () => {
+  if (!searchValue.value) return;
+
+  debouncedFetchAutocompleteResults('GET', {
+    params: { q: searchValue.value.toLowerCase() },
+  });
+};
+
+const handleLocationChange = (location: LocationInfo | null) => {
+  searchValue.value = location
+    ? `${location.localizedName}, ${location.country}`
+    : '';
+
+  emit('locationChange', location);
+};
+</script>
+
+<style scoped lang="scss">
+$dropdown-max-height: 240px;
+$input-width: 160px;
+$border-dropdown-list: solid 2px transparent;
```

Comment:
I don't see much sense in creating a variable that is only going to be used once. Also border width should go first.
`2px solid transparent`